lib/internal/modules/run_main.js

```diff
@@ -24,6 +24,10 @@ function shouldUseESMLoader(mainPath) {
   const userLoader = getOptionValue('--experimental-loader');
   if (userLoader)
     return true;
+  const esModuleSpecifierResolution =
+    getOptionValue('--es-module-specifier-resolution');
```

Comment:
Perfect. One thing to note is that this flag is experimental. I just posted an issue about this here - https://github.com/nodejs/node/issues/30600.

Since this is now the primary entry for this flag, we should probably move this check up to the top of the function, and also include a warning like:

```js
process.emitWarning(
    'The --es-module-specifier-resolution flag is experimental.',
    'ExperimentalWarning', undefined);
```

Would help to add this stuff while we're working on this area anyway, but if you'd rather it be done separately that is fine too just thought I'd mention as it's important for users not to overlook that any of these flags might be removed in future.

Reply:
Thanks for the explanation. I've noticed that #30600 is not only about `specifier-resolution` but also related to other experimental flags, so perhaps the fix about it should be raised in another PR rather than merged to this one.

Reply:
If I don't understand it wrong, I'll try to work on #30600 and create another PR to fix it.

Reply:
Sure if you're able to work on that it would be great, just mentioning as otherwise it can be useful to do these things while we're there.